aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -27,6 +35,12 @@ class ReduceMultiply {
   constexpr void operator() (scalar_t * self_data, scalar_t * src_data) const {
     *self_data *= *src_data;
   }
+  template <typename scalar_t,
+            typename std::enable_if_t<is_reduced_floating_point_v<scalar_t>, int> = 0>
+  constexpr void operator() (at::opmath_type<scalar_t> * self_data, scalar_t * src_data) const {
```

Comment:
this template should also cover the one above, right ?

in case this patch is landed, we don't need the specification of `tempalte <> operator<BFloat16, BFloat16>();` anymore.